lib/pbench/cli/server/shell.py

```diff
@@ -37,6 +40,57 @@ def find_the_unicorn(logger: Logger):
             )
 
 
+def keycloak_connection(config: PbenchServerConfig, logger: Logger):
+    """
+     Checks if the Keycloak server is up and accepting the connections.
+     The connection check does the GET request on the oidc server /health
+     endpoint and the sample response returned by the /health endpoint looks
+     like the following:
+        {
+            "status": "UP", # if the server is up
+            "checks": []
+        }
+     Note: The Keycloak server need to be started with health-enabled on.
+     Args:
+        config: PbenchServerConfig
+        logger: logger
+    Returns:
+        0 if successful
+    """
+    try:
+        oidc_server = str(config.get("authentication", "internal_server_url"))
+    except (NoSectionError):
+        logger.exception("Error fetching required configuration")
+        sys.exit(1)
+    except (NoOptionError):
+        try:
+            oidc_server = str(config.get("authentication", "server_url"))
+        except (NoOptionError):
+            logger.exception("Error fetching required configuration")
+            sys.exit(1)
+
+    # The connection check will run for a minute, and if the connection is
+    # successful it will break the look.
+    t_end = time.time() + 60
+    while time.time() <= t_end:
+        try:
+            response = requests.session().request(
+                "GET", f"{oidc_server}/health", verify=False
+            )
+            response.raise_for_status()
+            if response.json()["status"] == "UP":
+                return
+            else:
+                time.sleep(10)
+                continue
+        except Exception:
+            logger.exception("Exception while initializing OIDC client")
+            time.sleep(10)
```

Comment:
There's no need for the `else` statement nor the `continue` statement.  And, I would replace the two `sleep()` calls with a single one at the end of the loop (after the `try`/`except` block) -- there's no need to have separate delays for the two cases.